File: chatgpt-generated/python/idor.py
from flask import Flask, request

app = Flask(__name__)

fake_users = {
    '1': 'Alice',
    '2': 'Bob',
    '3': 'Charlie'
}

@app.route('/profile')
def profile():
    user_id = request.args.get('id')
    return f"User profile: {fake_users.get(user_id, 'Not found')}"

if __name__ == '__main__':
    app.run(debug=True)

File: chatgpt-generated/go/idor.go
package main

import (
	"fmt"
	"net/http"
)

var users = map[string]string{
	"1": "Alice",
	"2": "Bob",
	"3": "Charlie",
}

func main() {
	http.HandleFunc("/profile", func(w http.ResponseWriter, r *http.Request) {
		id := r.URL.Query().Get("id")
		fmt.Fprintf(w, "Profile: %s", users[id])
	})
	http.ListenAndServe(":8080", nil)
}
